lib/block_view.go

```diff
@@ -7558,43 +7573,38 @@ func (bav *UtxoView) _flushBalanceEntriesToDbWithTxn(txn *badger.Txn) error {
 }
 
 func (bav *UtxoView) FlushToDbWithTxn(txn *badger.Txn) error {
-	// Flush the utxos to the db.
-	if err := bav._flushUtxosToDbWithTxn(txn); err != nil {
-		return err
+	// Only flush to BadgerDB if Postgres is disabled
+	if bav.Postgres == nil {
```

Comment:
Hmm... so we only flush to db if Postgres is set?

Reply:
I think it's possibly better to have Postgres *in addition* to Badger initially, rather than trying to replace it. It looks like doing it this way makes you forced into using Postgres, whereas it may be better to leave it as optional until *everything* can be done in Postgres.

Reply:
Absolutely agree that Postgres is a totally bad idea to force to use it. Especially in the way we talk about the core project. I thought that by architecture backend project should operate data like indexes and API queries. But core stays in independent and lightweight mode and is easy to run by everyone who wants to get raw nodes and support network consensus (I hope we will see it soon, kind of POS or whatever).

Adding such a monster as Postgres in core I guess is a bad idea at all. It should be in the backend project. But it's just my opinion. Cheers.